### PR TITLE
add egs2/chime4/enh_asr1 recipe and results

### PR DESCRIPTION
**Pull request by simpleoier**
- [x] add README.md with results and pretrained models

### CODE REVIEW

File: egs2/chime4/enh_asr1/local/nn-gev [hunk 1]
```diff
@@ -0,0 +1 @@
+/jet/home/xuankaic/work/experiments/espnet/egs2/chime4/enh1/local/nn-gev
```
Comment (Emrys365) [inline egs2/chime4/enh_asr1/local/nn-gev:1]: Is this necessary?

**Review by Emrys365 — APPROVED** [review]
LGTM

File: egs2/chime4/enh_asr1/local/nn-gev (hunk 1 above)
Comment (simpleoier) [reply egs2/chime4/enh_asr1/local/nn-gev:1]: will remove this.

**Comment by Emrys365** [comment]
BTW, the argument `permutation_by_enh` in [espnet2/enh/espnet_enh_s2t_model.py](https://github.com/espnet/espnet/blob/master/espnet2/enh/espnet_enh_s2t_model.py#L35) seems unused.

File: egs2/chime4/enh_asr1/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+|decode_asr_transformer_normalize_output_wavtrue_lm_lm_train_lm_transformer_en_char_valid.loss.ave_enh_asr_model_valid.acc.ave/et05_real_beamformit_5mics|1320|126796|94.3|2.4|3.3|1.0|6.6|61.2|
+|decode_asr_transformer_normalize_output_wavtrue_lm_lm_train_lm_transformer_en_char_valid.loss.ave_enh_asr_model_valid.acc.ave/et05_real_isolated_1ch_track|1320|126796|91.5|3.8|4.6|1.6|10.0|69.4|
+|decode_asr_transformer_normalize_output_wavtrue_lm_lm_train_lm_transformer_en_char_valid.loss.ave_enh_asr_model_valid.acc.ave/et05_simu_beamformit_2mics|1320|126812|92.2|3.5|4.2|1.7|9.5|67.5|
+|decode_asr_transformer_normalize_output_wavtrue_lm_lm_train_lm_transformer_en_char_valid.loss.ave_enh_asr_model_valid.acc.ave/et05_simu_beamformit_5mics|1320|126812|93.7|2.7|3.5|1.4|7.7|64.8|
+|decode_asr_transformer_normalize_output_wavtrue_lm_lm_train_lm_transformer_en_char_valid.loss.ave_enh_asr_model_valid.acc.ave/et05_simu_isolated_1ch_track|1320|126812|90.3|4.8|4.9|2.2|11.9|69.4|
+
+### TER
+
+|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
+|---|---|---|---|---|---|---|---|---|
```
Comment (Emrys365) [inline egs2/chime4/enh_asr1/README.md:50]: Maybe this can be removed.

File: egs2/chime4/enh_asr1/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+### Enhancement
+
+|dataset|STOI|SAR|SDR|SI_SNR|
+|---|---|---|---|---|
+|dt05_simu_isolated_1ch_track|0.87|7.14|7.14|4.51|
+|et05_simu_isolated_1ch_track|0.85|7.47|7.47|3.02|
```
Comment (Emrys365) [inline egs2/chime4/enh_asr1/README.md:57]: You can remove the SAR column.

File: egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+]
+init_param: [
+    "../enh1/exp/enh_train_enh_convtasnet_small_raw/valid.loss.ave_1best.pth:encoder:enh_model.encoder",
+    "../enh1/exp/enh_train_enh_convtasnet_small_raw/valid.loss.ave_1best.pth:separator:enh_model.separator",
+    "../enh1/exp/enh_train_enh_convtasnet_small_raw/valid.loss.ave_1best.pth:decoder:enh_model.decoder",
+    "../asr1/exp/asr_train_asr_transformer_wavlm_lr1e-3_specaug_accum1_preenc128_warmup20k_raw_en_char/valid.acc.ave.pth:frontend:s2t_model.frontend",
+    "../asr1/exp/asr_train_asr_transformer_wavlm_lr1e-3_specaug_accum1_preenc128_warmup20k_raw_en_char/valid.acc.ave.pth:preencoder:s2t_model.preencoder",
+    "../asr1/exp/asr_train_asr_transformer_wavlm_lr1e-3_specaug_accum1_preenc128_warmup20k_raw_en_char/valid.acc.ave.pth:encoder:s2t_model.encoder",
+    "../asr1/exp/asr_train_asr_transformer_wavlm_lr1e-3_specaug_accum1_preenc128_warmup20k_raw_en_char/valid.acc.ave.pth:ctc:s2t_model.ctc",
+    "../asr1/exp/asr_train_asr_transformer_wavlm_lr1e-3_specaug_accum1_preenc128_warmup20k_raw_en_char/valid.acc.ave.pth:decoder:s2t_model.decoder",
```
Comment (Emrys365) [inline egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml:34]: Maybe it's better to include the config files for pretraining these models.

**Review by Emrys365 — APPROVED** [review]

File: egs2/chime4/enh_asr1/README.md (hunk 2 above)
Comment (simpleoier) [reply egs2/chime4/enh_asr1/README.md]: done.

File: egs2/chime4/enh_asr1/README.md (hunk 3 above)
Comment (simpleoier) [reply egs2/chime4/enh_asr1/README.md]: done

File: egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml (hunk 4 above)
Comment (simpleoier) [reply egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml:34]: done

**Comment by simpleoier** [comment]
@sw005320 I think this PR is ready to be merged. Can you take a look?

File: egs2/chime4/enh_asr1/README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+- Git hash: `44971ff962aae30c962226f1ba3d87de057ac00e`
+  - Commit date: `Wed Apr 27 10:13:03 2022 -0400`
+
+## enh_asr_train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0_raw_en_char
+- Pretrained model: https://huggingface.co/espnet/simpleoier_chime4_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_raw_en_char
+### WER
+
+|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
+|---|---|---|---|---|---|---|---|---|
+|decode_asr_transformer_normalize_output_wavtrue_lm_lm_train_lm_transformer_en_char_valid.loss.ave_enh_asr_model_valid.acc.ave/dt05_real_beamformit_2mics|1640|27119|98.5|1.2|0.3|0.2|1.7|19.6|
```
Comment (sw005320) [inline egs2/chime4/enh_asr1/README.md:17]: Can you re-order it with 1ch --> 2ch --> 5ch?

File: egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+-   - train
+    - loss
+    - min
+keep_nbest_models: 10
+num_att_plot: 3
+unused_parameters: true
+freeze_param: [
+    "s2t_model.frontend.upstream",
+]
+init_param: [
```
Comment (sw005320) [inline egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml:26]: Can you add a comment about the strategy of freezing parameters?

**Review by sw005320 — COMMENTED** [review]
Some minor comments.

File: egs2/chime4/enh_asr1/README.md (hunk 5 above)
Comment (simpleoier) [reply egs2/chime4/enh_asr1/README.md:18]: OK.

File: egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml (hunk 6 above)
Comment (simpleoier) [reply egs2/chime4/enh_asr1/conf/tuning/train_enh_asr_convtasnet_init_noenhloss_wavlm_transformer_init_lr1e-4_accum1_adam_specaug_bypass0.yaml:26]: Yeah, just added it.

**Pull request merged**